elastic/eui
# [EuiHeader] Update to set a new global `--euiFixedHeadersOffset` CSS variable

## Summary

![screencap](https://github.com/elastic/eui/assets/549407/fe4a2fb1-cdfa-4cc6-b047-f5bfea5fd777)

This logic allows `EuiHeader` to afford for any number of fixed headers (although it's likely that consumers will not use more than two). The most important aspect of this refactor that other EUI components can access this CSS variable globally with no extra JS logic or rerendering on their end, which removes a significant amount of duplicated `useEffects` from components affected by fixed headers (`EuiPageTemplate`, `EuiSideBar`, and `EuiCollapsibleNavBeta`), and for other components (`EuiFlyout` and `EuiOverlayMask`), it allows them to correctly and dynamically set their top position and height instead of relying on a static Sass mixin to work as expected.

I strongly recommend [reviewing by commit](https://github.com/elastic/eui/pull/7144/commits) as multiple downstream components/usages were updated separately.

## QA

- `gh pr checkout 7144 && yarn storybook`
- Go to http://localhost:6006/?path=/story/euiheader--multiple-fixed-headers
- [x] Dynamically add and remove headers, and confirm that each time, the page template and flyout mask correctly accounts for the number of fixed headers on the page, including when that number is 0
- Go to https://eui.elastic.co/pr_7144/#/layout/header#fixed-header
- [x] Confirm that toggling the demo header appends the fixed header below the existing docs header

### General checklist
